src/pudl/transform/eia860m.py

```diff
@@ -120,19 +120,25 @@ def core_eia860m__changelog_generators(
     eia860m_all["report_date_max"] = eia860m_all.groupby(gen_idx_no_date)[
         "report_date"
     ].transform("max")
-    # drop duplicates after sorting by date so we get the first appreance
+    # drop duplicates after sorting by date so we get the first appearance
     eia860m_changelog = eia860m_all.sort_values(
         by=["report_date"], ascending=True
     ).drop_duplicates(
         subset=[c for c in eia860m_all if c != "report_date"],
         keep="first",
     )
 
-    eia860m_changelog["valid_until_date"] = (
+    report_date_max_mask = (
+        eia860m_changelog["report_date"] == eia860m_changelog["report_date_max"]
+    )
+    eia860m_changelog.loc[~report_date_max_mask, "valid_until_date"] = (
         eia860m_changelog.sort_values(gens_idx, ascending=False)
         .groupby(gen_idx_no_date)["report_date"]
         .transform("shift")
         .fillna(eia860m_changelog.report_date_max)
```

Comment:
It skeeved me out to see value assignment to a differently-sorted and different length series, but it seems to work just fine. I guess pandas must match on the index.